experimental/FTheoryTools/src/auxiliary.jl

```diff
@@ -170,51 +170,115 @@ function _kodaira_type(id::MPolyIdeal{T}, f::T, g::T, d::T, ords::Tuple{Int64, I
-      g_quotient = divrem(9 * poly_g, locus)[2]
-      f_quotient = divrem(2 * poly_f, locus)[2]
-      quotient_val = div(g_quotient, f_quotient)
-
-      monodromy_poly = _psi^2 + quotient_val
-      kod_type = _string_from_factor_count(monodromy_poly, ["Non-split I_$d_ord", "Split I_$d_ord"])
-    elseif d_ord == 4 && g_ord == 2 && f_ord >= 2
-      g_quotient = divrem(div(poly_g, locus^2), locus)[2]
-
-      monodromy_poly = _psi^2 - g_quotient
-      kod_type = _string_from_factor_count(monodromy_poly, ["Non-split IV", "Split IV"])
-    elseif d_ord == 6 && f_ord >= 2 && g_ord >= 3
-      f_quotient = divrem(div(poly_f, locus^2), locus)[2]
-      g_quotient = divrem(div(poly_g, locus^3), locus)[2]
-      
-      monodromy_poly = _psi^3 + _psi * f_quotient + g_quotient
-      kod_type = _string_from_factor_count(monodromy_poly, ["Non-split I^*_0", "Semi-split I^*_0", "Split I^*_0"])
-    elseif f_ord == 2 && g_ord == 3 && d_ord >= 7
-      d_quotient = div(poly_d, locus^d_ord)
-      f_quotient = div(2 * poly_f, locus^2)
-      g_quotient = div(9 * poly_g, locus^3)
-      num_quotient = divrem(d_quotient * f_quotient^(2 + d_ord % 2), locus)[2]
-      den_quotient = divrem(4 * g_quotient^(2 + d_ord % 2), locus)[2]
-      quotient_val = div(num_quotient, den_quotient)
-
-      monodromy_poly = _psi^2 + quotient_val
-      kod_type = _string_from_factor_count(monodromy_poly, ["Non-split I^*_$(d_ord - 6)", "Split I^*_$(d_ord - 6)"])
-    elseif d_ord == 8 && g_ord == 4 && f_ord >= 3
-      g_quotient = divrem(div(poly_g, locus^4), locus)[2]
-
-      monodromy_poly = _psi^2 - g_quotient
-      kod_type = _string_from_factor_count(monodromy_poly, ["Non-split IV^*", "Split IV^*"])
+    if concrete_base
+      # Over concrete bases, we reduce to only two variables
+      # so that the computation is faster
+      num_gens = length(gens(parent(f)))
+      first_coord_inds = collect(1:num_gens - 2)
+      last_coord_inds = collect(3:num_gens)
+      rand_ints_ab2 = rand(-100:100, num_gens - 2)
+      w_first_2 = evaluate(forget_decoration(gens(id)[1]), last_coord_inds, rand_ints_ab2)
+      f_first_2 = evaluate(forget_decoration(f), last_coord_inds, rand_ints_ab2)
+      g_first_2 = evaluate(forget_decoration(g), last_coord_inds, rand_ints_ab2)
+      d_first_2 = evaluate(forget_decoration(d), last_coord_inds, rand_ints_ab2)
+      w_last_2 = evaluate(forget_decoration(gens(id)[1]), first_coord_inds, rand_ints_ab2)
+      f_last_2 = evaluate(forget_decoration(f), first_coord_inds, rand_ints_ab2)
+      g_last_2 = evaluate(forget_decoration(g), first_coord_inds, rand_ints_ab2)
+      d_last_2 = evaluate(forget_decoration(d), first_coord_inds, rand_ints_ab2)
+
+      # Check monodromy conditions for remaining cases
+      if f_ord == 0 && g_ord == 0
+        q_first = quotient(ideal([9 * g_first_2, w_first_2]), ideal([2 * f_first_2, w_first_2]))
+        q_last = quotient(ideal([9 * g_last_2, w_last_2]), ideal([2 * f_last_2, w_last_2]))
+
```

Comment:
Just for my curiosity: Why is there a line break here (and similarly below)? I am not sure if it enhances readability.

Reply:
Only for readability. I generally prefer to break up the logical steps of the calculation rather than have a solid block of code. For some of these cases, the first step is many lines, and for consistency I line break even the shorter cases